src/main/java/org/deepsymmetry/beatlink/SocketSender.java

```diff
@@ -0,0 +1,33 @@
+package org.deepsymmetry.beatlink;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.util.concurrent.atomic.AtomicReference;
+
+public interface SocketSender {
+
+    /**
+     * Convenience method to wrap the socket sending so that other classes can use it.
+     * This does not need to be synchronized because the whole packet is sent to the OS
+     * to be queued to send, safe to send from many threads.
+     *
+     * @param packet sends packet to socket connection to be queued by the OS.
+     * @throws IOException Throws this if the socket can't connect or has failed in another way
+     */
+    void send(DatagramPacket packet) throws IOException;
+
+    /**
+     * A method for socket senders to provide access to the underlying socket, but only
+     * for sending reasons. If a process wants to receive from a socket, they must register
+     * a listener.
+     *
+     * @return SocketSender
+     */
+    SocketSender getSocketSender();
+
+    InetAddress getLocalAddress();
+
+    void close();
```

Comment:
And we should document this, if we keep it. Is there any reason to? I assume any of the classes that implement this interface will also need to be `LifecycleParticipant` implementations that already have `start()` and `stop()` methods. And I see no reason they should let outside classes close their socket on them.